src/workerd/api/url.h

```diff
@@ -217,21 +213,28 @@ class URLSearchParams: public jsg::Object {
 private:
   kj::Own<URL::RefcountedUrl> url;
 
-  template <typename Type>
-  static kj::Maybe<Type> iteratorNext(jsg::Lock& js, IteratorState& state) {
+  static kj::Maybe<kj::Array<kj::String>> entryIteratorNext(jsg::Lock& js, IteratorState& state) {
     if (state.index >= state.parent->url->query.size()) {
       return nullptr;
     }
     auto& [key, value] = state.parent->url->query[state.index++];
-    if constexpr (kj::isSameType<Type, EntryIteratorType>()) {
-      return kj::arr(kj::str(key), kj::str(value));
-    } else if constexpr (kj::isSameType<Type, KeyIteratorType>()) {
-      return kj::str(key);
-    } else if constexpr (kj::isSameType<Type, ValueIteratorType>()) {
```

Comment:
@harrishancock ... the bug was here. `KeyIteratorType` and `ValueIteratorType` were the same type. doh.

Reply:
Oh, wow.